app/addons/documents/mango/components/ExecutionStats.js

```diff
@@ -0,0 +1,127 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+import React from 'react';
+import { Popover, OverlayTrigger } from 'react-bootstrap';
+
+const TOO_MANY_DOCS_SCANNED_WARNING = "The number of documents examined is high in proportion to the number of results returned. Consider adding an index to improve this.";
+
+export default class ExecutionStats extends React.Component {
+  constructor (props) {
+    super(props);
+  }
+
+  humanizeDuration(milliseconds) {
+    if (milliseconds < 1000) {
+      return Math.round(milliseconds) + ' ms';
+    }
+    let seconds = milliseconds / 1000;
+    if (seconds < 60) {
+      return Math.floor(seconds) + ' seconds';
+    }
+    const minutes = Math.floor(seconds / 60);
+    seconds = seconds - (minutes * 60);
+
+    return minutes + 'minute, ' + seconds + 'seconds';
+  }
+
+  getWarning(executionStats, warning) {
+    if (executionStats) {
+      // warn if many documents scanned in relation to results returned
+      if (executionStats && executionStats.results_returned) {
```

Comment:
This can be `if (executionStats.results_returned) {`

Reply:
Actually you can invert the first if statement:

`if (!executionStats) { return warning;}`